stake-pool/py/stake_pool/actions.py

```diff
@@ -589,3 +590,56 @@ async def decrease_validator_stake(
     signers = [payer, staker] if payer != staker else [payer]
     await client.send_transaction(
         txn, *signers, opts=TxOpts(skip_confirmation=False, preflight_commitment=Confirmed))
+
+
+async def create_token_metadata(client: AsyncClient, payer: Keypair, stake_pool_address: PublicKey,
+                                name: str, symbol: str, uri: str):
+    resp = await client.get_account_info(stake_pool_address, commitment=Confirmed)
+    data = resp['result']['value']['data']
+    stake_pool = StakePool.decode(data[0], data[1])
+
+    (withdraw_authority, _seed) = find_withdraw_authority_program_address(STAKE_POOL_PROGRAM_ID, stake_pool_address)
+
+    txn = Transaction()
+    txn.add(
+        sp.CreateTokenMetadataParams(
```

Comment:
Maybe I'm missing something, but this just gives a tuple, and not an instruction. You'll need to call `create_metadata_account` with `sp.CreateTokenMetadataParams` as a parameter.

Reply:
Yep sorry, missed it